### ISSUE TITLE
Clang crashed when used [[no_unique_address]].

### ISSUE THREAD

**Issue by Adttil** [llvm/llvm-project#104227]
See on godbolt : [https://gcc.godbolt.org/z/68ET15Thn](https://gcc.godbolt.org/z/68ET15Thn)
This simple code make clang18 crashed. It seems to be related to `[[no_unique_address]]`.   
Removing `[[no_unique_address]]` from `wrapper` or adding `[[no_unique_address]]` to `foo` as stated in the comments will result in successful compilation.  
```cpp
#include <type_traits>
#include <utility>

struct E{};

struct wrapper{
    [[no_unique_address]] // delete this
    E value{};
};

struct foo{
    //[[no_unique_address]] // or add this
    wrapper value{};

    constexpr auto fn()const{
        return 0;
    }
};

int main(){
    auto a = foo{}.fn();
}
```

**Comment by Adttil** [comment]
Sorry，I just noticed a similar closed issue：[#89547](https://github.com/llvm/llvm-project/issues/89547).  
It has been closed as completed in [#90303](https://github.com/llvm/llvm-project/pull/90303).
@efriedma-quic  
Please check if this merge also solves this problem. Thanks!

**Comment by Adttil** [comment]
Sorry，I just noticed a similar closed issue：https://github.com/llvm/llvm-project/issues/89547.
It has been closed as completed in https://github.com/llvm/llvm-project/pull/90303.
And I test this code with clang(trunk)，It can already be compiled successfully.
